src/connection-monitor-preview/azext_connection_monitor_preview/_params.py

```diff
@@ -0,0 +1,304 @@
+from knack.arguments import CLIArgumentType, ignore_type
+from azure.cli.core.commands.parameters import (get_enum_type,
+                                                get_location_type,
+                                                get_three_state_flag)
+
+from ._validators import (get_network_watcher_from_location,
+                          NWConnectionMonitorEndpointFilterItemAction,
+                          NWConnectionMonitorTestConfigurationHTTPRequestHeaderAction)
+
+
+def load_arguments(self, _):
+    name_arg_type = CLIArgumentType(options_list=['--name', '-n'], metavar='NAME')
+
+    (ConnectionMonitorEndpointFilterType, ConnectionMonitorTestConfigurationProtocol,
+     PreferredIPVersion, HTTPConfigurationMethod,
+     OutputType) = self.get_models(
+         'ConnectionMonitorEndpointFilterType', 'ConnectionMonitorTestConfigurationProtocol',
+         'PreferredIPVersion', 'HTTPConfigurationMethod',
+         'OutputType')
+
+    with self.argument_context('network watcher') as c:
+        c.argument('network_watcher_name', name_arg_type, help='Name of the Network Watcher.')
+        c.argument('location', validator=None)
+        c.ignore('watcher_rg')
+        c.ignore('watcher_name')
+
+    with self.argument_context('network watcher connection-monitor') as c:
+        c.argument('network_watcher_name', arg_type=ignore_type, options_list=['--__NETWORK_WATCHER_NAME'])
+        c.argument('connection_monitor_name', name_arg_type, help='Connection monitor name.')
+
+    with self.argument_context('network watcher connection-monitor', arg_group='V1 Source') as c:
+        c.argument('source_resource', help='Name or ID of the resource from which to originate traffic. '
+                                           'Currently only Virtual Machines are supported.')
+        c.argument('source_port', help='Port number from which to originate traffic.')
+
+    with self.argument_context('network watcher connection-monitor', arg_group='V1 Destination') as c:
+        c.argument('dest_resource', help='Name of ID of the resource to receive traffic. '
+                                         'Currently only Virtual Machines are supported.')
+        c.argument('dest_port', help='Port number on which to receive traffic.')
+        c.argument('dest_address', help='The IP address or URI at which to receive traffic.')
+
+    nw_validator = get_network_watcher_from_location(remove=True, watcher_name='network_watcher_name',
+                                                     rg_name='resource_group_name')
+    for scope in ['list', 'show', 'start', 'stop', 'delete', 'query']:
+        with self.argument_context('network watcher connection-monitor {}'.format(scope)) as c:
+            c.extra('location', get_location_type(self.cli_ctx), required=True)
+            c.argument('resource_group_name', arg_type=ignore_type, validator=nw_validator)
+
+    with self.argument_context('network watcher connection-monitor create', arg_group='V1') as c:
```

Comment:
should it be `arg_group='V1 Endpoint'`?

Reply:
I re-group them,
V1 Endpoint Arguemnt incldues: `--source-xxxx`, `--dest-xxx`, `--do-not-start`, `--monitoring-interval`
V2 Argument includes only: `--notes`

Reply:
I re-group them:
![image](https://user-images.githubusercontent.com/14357159/73659574-7e579680-46d1-11ea-98db-3faf57ad5c48.png)